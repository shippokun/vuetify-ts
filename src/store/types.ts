export interface RootState {
  version: string;
}

export interface DrawState {
  drawer: boolean;
}
