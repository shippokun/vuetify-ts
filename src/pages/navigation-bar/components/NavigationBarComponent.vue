<template>
  <v-app-bar app color="indigo" dark>
    <v-app-bar-nav-icon @click.stop="emitToggle()"></v-app-bar-nav-icon>
    <v-toolbar-title>Application</v-toolbar-title>
  </v-app-bar>
</template>

<script lang="ts">
import { Component, Prop, Emit, Vue } from 'vue-property-decorator';

@Component
export default class NavigationBarComponent extends Vue {
  @Prop()
  public drawer!: boolean;

  @Emit()
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  public emit(value: boolean) {}

  public emitToggle(): void {
    this.emit(!this.drawer);
  }
}
</script>
