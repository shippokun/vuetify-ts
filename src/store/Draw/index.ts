import { Module } from 'vuex';
import { DrawState, RootState } from '../types';
import actions from './actions';
import mutations from './mutations';

const state: DrawState = {
  drawer: false,
};

export const draw: Module<DrawState, RootState> = {
  namespaced: true,
  state,
  actions,
  mutations,
};
