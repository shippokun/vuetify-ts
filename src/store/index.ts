import Vuex, { StoreOptions } from 'vuex';
import { RootState } from './types';

const store: StoreOptions<RootState> = {
  state: {
    version: '1.0.0',
  },
  mutations: {},
  actions: {},
  modules: {},
};

export default new Vuex.Store<RootState>(store);
