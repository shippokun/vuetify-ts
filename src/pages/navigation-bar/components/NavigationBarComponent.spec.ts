import { shallowMount, createLocalVue } from '@vue/test-utils';
import NavigationBarComponent from './NavigationBarComponent.vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
import { DrawState } from '../../../store/types';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('NavigationBarComponent.vue', () => {
  let vuetify: any;
  let store: any;
  let state: DrawState;

  beforeEach(() => {
    vuetify = new Vuetify({
      mocks: {
        breakpoint: {
          mobileBreakpoint: 0,
        },
      },
    });
    store = new Vuex.Store({
      modules: {
        draw: {
          namespaced: true,
          state: {
            drawer: false,
          },
        },
      },
    });
  });
  it('snapshots', () => {
    const wrapper = shallowMount(NavigationBarComponent, {
      vuetify,
      store,
      localVue,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  describe('computed', () => {
    it('set drawer', () => {
      const wrapper = shallowMount(NavigationBarComponent, {
        vuetify,
        localVue,
        store,
        propsData: {
          toggleDraw: jest.fn(),
        },
      });
      const mockInput = true;
      // @ts-ignore
      wrapper.vm.drawer = mockInput;

      // @ts-ignore
      expect(wrapper.vm.toggleDraw).toBeCalledWith(mockInput);
    });
  });
  describe('methods', () => {
    it('emitToggle', () => {
      const wrapper = shallowMount(NavigationBarComponent, {
        vuetify,
        localVue,
        store,
        propsData: {
          toggleDraw: jest.fn(),
        },
      });
      const mockInput = false;
      // @ts-ignore
      wrapper.vm.emitToggle(mockInput);

      // @ts-ignore
      expect(wrapper.vm.toggleDraw).toBeCalledWith(mockInput);
    });
  });
});
