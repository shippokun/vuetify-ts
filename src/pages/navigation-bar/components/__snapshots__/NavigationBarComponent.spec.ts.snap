// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBarComponent.vue snapshots 1`] = `
<v-navigation-drawer-stub app="true" mobilebreakpoint="1264" height="100vh" minivariantwidth="56" src="" tag="nav" width="256">
  <v-list-stub tag="div" dense="true">
    <v-list-item-stub activeclass="" link="true" tag="div">
      <v-list-item-action-stub>
        <v-icon-stub>mdi-home</v-icon-stub>
      </v-list-item-action-stub>
      <v-list-item-content-stub>
        <v-list-item-title-stub>Home</v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub activeclass="" link="true" tag="div">
      <v-list-item-action-stub>
        <v-icon-stub>mdi-email</v-icon-stub>
      </v-list-item-action-stub>
      <v-list-item-content-stub>
        <v-list-item-title-stub>Contact</v-list-item-title-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
  </v-list-stub>
</v-navigation-drawer-stub>
`;
